index.js

```diff
@@ -270,10 +270,13 @@ SonyTV.prototype.receiveApplications = function() {
       if (data.indexOf("error") < 0){
         var jayons = JSON.parse(data);
         var reslt = jayons.result[0];
-        reslt.forEach(function(source){
-          if (that.applications.length == 0 || that.applications.indexOf(source.title) >= 0) {
+        reslt.sort(source => source.title).forEach(function(source){
+          if (that.applications.length == 0 || that.applications.map(app => app.title).filter(title => source.title.includes(title)).length > 0) {
```

Comment:
There's an opportunity for a minor performance improvement here by moving the `that.applications.map(...)` up to a variable outside the `foreach` so it isn't executed for each app returned by the TV.